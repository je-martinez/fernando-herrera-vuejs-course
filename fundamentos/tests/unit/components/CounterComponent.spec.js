import { shallowMount } from "@vue/test-utils";
import Counter from "@/components/CounterComponent.vue";

describe("Counter.vue", () => {
  const wrapper = shallowMount(Counter);
  it("renders props.msg be default value", () => {
    // const msg = undefined;
    // const wrapper = shallowMount(Counter, {
    //   props: {
    //     title: msg,
    //   },
    // });
    const title = wrapper.find("h2");
    expect(title.exists()).toBeTruthy();
    expect(title.text()).toBe("Counter");
  });

  it("should second <p> tag have default value", () => {
    // const wrapper = shallowMount(Counter);

    // const texts = wrapper.findAll("p");
    // expect(texts[1].text()).toBe("0");

    const text = wrapper.find(`[data-testid="counter"]`);
    expect(text.text()).toBe("0");
  });

  it("should increment the counter", async () => {
    // const wrapper = shallowMount(Counter);
    const incrementButton = wrapper.find("#increment-btn");
    await incrementButton.trigger("click");
    const text = wrapper.find(`[data-testid="counter"]`);
    expect(text.text()).toBe("1");
  });

  it("should decrement the counter", async () => {
    // const wrapper = shallowMount(Counter);
    const decrementButton = wrapper.find("#decrement-btn");
    await decrementButton.trigger("click");
    const text = wrapper.find(`[data-testid="counter"]`);
    expect(text.text()).toBe("0");
  });
});
