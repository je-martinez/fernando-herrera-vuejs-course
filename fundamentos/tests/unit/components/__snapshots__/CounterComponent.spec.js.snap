// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue renders props.title when passed: new message 1`] = `
<h2>new message</h2>
<p>0 <sup>2</sup> = 0</p>
<p data-testid="counter">0</p>
<div><button id="increment-btn">+1</button><button id="decrement-btn">-1</button></div>
`;
